### PR TITLE
plugins: fix python shebang

### PR DESCRIPTION
**Pull request by tigarmo**
Use ${PARTS_PYTHON_INTERPRETER} to play nice with the upstream plugin;
If the user overrides this, there's probably a good reason (like using
Python 3.6).

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -18,7 +18,8 @@ commonmark==0.9.1
 coverage==6.5.0
 craft-archives==1.1.0
 craft-cli==2.0.0
-craft-parts==1.22.0
+#craft-parts==1.22.0
+git+https://github.com/tigarmo/craft-parts.git@fix-python-3.6#egg=craft_parts
```
Comment (tigarmo) [inline requirements-dev.txt:22]: self-reminder to update this

**Review by lengau — APPROVED** [review]
Looks good! I gather this is going to need a craft-parts 1.23 update?

File: tests/spread/general/plugin-python-3.6/rockcraft.orig.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    priority: always
+
+parts:
+    my-part:
+        plugin: python
+        source: .
+        stage-packages: [python3.6-venv]
+        python-packages: [black]
+        build-environment:
+            - PARTS_PYTHON_INTERPRETER: python3.6
```
Comment (cmatsuoka) [inline tests/spread/general/plugin-python-3.6/rockcraft.orig.yaml:23]: What happens if the user tries to specify the full path for the interpreter? If that's not allowed, maybe we should detect and warn? Or, if that's valid, the shebang line shouldn't hardcode `/bin/`.

Comment (tigarmo) [reply tests/spread/general/plugin-python-3.6/rockcraft.orig.yaml:23]: Hmm good question. This `PARTS_PYTHON_INTERPRETER` variable is used both during build and during runtime (in the generated containers) and I'm having a hard time thinking of a full path value that would work in both "stages", so I think we should block it. I'll add a check and a test

File: rockcraft/plugins/python_plugin.py [hunk 3]
```diff
@@ -86,7 +86,7 @@ def _get_system_python_interpreter(self) -> Optional[str]:
     @override
     def _get_script_interpreter(self) -> str:
         """Overridden because Python is always available in /bin/python3."""
```
[the remark this answers is not in the document]
Comment (tigarmo) [reply rockcraft/plugins/python_plugin.py]: you're right! updated

File: tests/spread/general/plugin-python-3.6/rockcraft.orig.yaml (hunk 2 above)
Comment (tigarmo) [reply tests/spread/general/plugin-python-3.6/rockcraft.orig.yaml:23]: Added check & test

File: rockcraft/plugins/python_plugin.py [hunk 4]
```diff
@@ -85,20 +85,35 @@ def _get_system_python_interpreter(self) -> Optional[str]:
...
-        original = super().get_build_commands()
+        commands = []
+
+        # Detect whether PARTS_PYTHON_INTERPRETER is a full path (not supported)
+        commands.append(
+            dedent(
+                """
+                # Detect whether PARTS_PYTHON_INTERPRETER is a full path
+                if [[ "${PARTS_PYTHON_INTERPRETER}" = /* ]]; then
+                    echo "Full paths in \"PARTS_PYTHON_INTERPRETER\" are not allowed: ${PARTS_PYTHON_INTERPRETER}"
```
Comment (cmatsuoka) [inline rockcraft/plugins/python_plugin.py:102]: Maybe absolute paths instead of full paths?

**Review by cmatsuoka — APPROVED** [review]
Thanks! Added a note about path naming consistency, I think it's referred to as absolute path elsewhere.

File: rockcraft/plugins/python_plugin.py (hunk 4 above)
Comment (tigarmo) [reply rockcraft/plugins/python_plugin.py:102]: argh you are so right. I'll fix

**Pull request merged**